src/pipelines/vaccinations/ch_authority.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Any, Dict, List
+import requests
+from pandas import DataFrame
+from lib.data_source import DataSource
+from lib.utils import table_rename
+
+
+class SwitzerlandDataSource(DataSource):
+    def fetch(
+            self,
+            output_folder: Path,
+            cache: Dict[str, str],
+            fetch_opts: List[Dict[str, Any]],
+            skip_existing: bool = False,
+    ) -> Dict[str, str]:
+        # the url in the config is a json file which contains the actual url for vaccDosesAdministered
+        src_url = fetch_opts[0]['url']
+        data = requests.get(src_url).json()
+        vaccines_url = data['sources']['individual']['csv']['vaccDosesAdministered']
+        fetch_opts[0]['url'] = vaccines_url
+        return super().fetch(output_folder, cache, fetch_opts, skip_existing)
+
+    def parse_dataframes(
+        self, dataframes: Dict[Any, DataFrame], aux: Dict[str, DataFrame], **parse_opts
+    ) -> DataFrame:
+
+        # Rename the appropriate columns
+        data = table_rename(
+            dataframes[0],
+            {
+                "date": "date",
+                "geoRegion": "subregion1_code",
+                "sumTotal": "total_vaccine_doses_administered",
```

Comment:
The website shows total vaccine doses and total vaccinated people. Can you find total vaccinated people anywhere in the API?

Reply:
https://www.covid19.admin.ch/api/data/context also has "fullyVaccPersons": "https://www.covid19.admin.ch/api/data/20210217-9y1lrj19/sources/COVID19FullyVaccPersons.csv"

this however is only latest. only contains 2021-02-14. I guess for this I will have to use a cache. lets tackle in a follow-up?

Reply:
Sounds good. We can try reaching out to them as well to see if they plan on adding the historical values to that API call too.